modules/core/src/main/java/org/apache/ignite/internal/managers/GridManagerAdapter.java

```diff
@@ -256,27 +256,6 @@ protected final void startSpi() throws IgniteCheckedException {
                 inject(spi);
             }
 
-            try {
-                Map<String, Object> retval = spi.getNodeAttributes();
-
-                if (retval != null) {
-                    for (Map.Entry<String, Object> e : retval.entrySet()) {
-                        if (ctx.hasNodeAttribute(e.getKey()))
-                            throw new IgniteCheckedException("SPI attribute collision for attribute [spi=" + spi +
-                                ", attr=" + e.getKey() + ']' +
-                                ". Attribute set by one SPI implementation has the same name (name collision) as " +
-                                "attribute set by other SPI implementation. Such overriding is not allowed. " +
-                                "Please check your Ignite configuration and/or SPI implementation to avoid " +
-                                "attribute name collisions.");
-
-                        ctx.addNodeAttribute(e.getKey(), e.getValue());
-                    }
-                }
-            }
-            catch (IgniteSpiException e) {
-                throw new IgniteCheckedException("Failed to get SPI attributes.", e);
-            }
-
             // Print-out all SPI parameters only in DEBUG mode.
             if (log.isDebugEnabled())
                 log.debug("Starting SPI: " + spi);
```

Comment:
Would you declare 'names' a Set instead of collection? 
This allow to simplify boilerplate 'names.contains' then 'names.add' code.